pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -0,0 +1,327 @@
+			return allErrs
+		}
+
+		// Convert old gardenlet config to an internal version
+		oldGardenletConfig, err := helper.ConvertGardenletConfig(oldGardenlet.Config)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(configPath, oldGardenlet.Config, fmt.Sprintf("could not convert gardenlet config: %v", err)))
+			return allErrs
+		}
+
+		// Validate gardenlet config update
+		allErrs = append(allErrs, validateGardenletConfigurationUpdate(newGardenletConfig, oldGardenletConfig, configPath)...)
+	}
+
+	// Ensure bootstrap is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.Bootstrap, oldGardenlet.Bootstrap, fldPath.Child("bootstrap"))...)
+
+	// Ensure merge with parent is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.MergeWithParent, oldGardenlet.MergeWithParent, fldPath.Child("mergeWithParent"))...)
+
+	return allErrs
+}
+
+func validateGardenletDeployment(deployment *seedmanagement.GardenletDeployment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if deployment.ReplicaCount != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*deployment.ReplicaCount), fldPath.Child("replicaCount"))...)
+	}
+	if deployment.RevisionHistoryLimit != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*deployment.RevisionHistoryLimit), fldPath.Child("revisionHistoryLimit"))...)
+	}
+
+	if deployment.Image != nil {
+		allErrs = append(allErrs, validateImage(deployment.Image, fldPath.Child("image"))...)
+	}
+
+	allErrs = append(allErrs, metav1validation.ValidateLabels(deployment.PodLabels, fldPath.Child("podLabels"))...)
+	allErrs = append(allErrs, apivalidation.ValidateAnnotations(deployment.PodAnnotations, fldPath.Child("podAnnotations"))...)
+
+	return allErrs
+}
+
+func validateImage(image *seedmanagement.Image, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if image.PullPolicy != nil {
+		validValues := []string{string(corev1.PullAlways), string(corev1.PullIfNotPresent), string(corev1.PullNever)}
+		if !utils.ValueExists(string(*image.PullPolicy), validValues) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("pullPolicy"), *image.PullPolicy, validValues))
+		}
+	}
+
+	return allErrs
+}
+
+func validateGardenletConfiguration(gardenletConfig *config.GardenletConfiguration, bootstrap seedmanagement.Bootstrap, mergeWithParent bool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Ensure seed selector is not specified
+	if gardenletConfig.SeedSelector != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedSelector"), "seed selector is forbidden"))
+	}
+
+	// Validate gardenlet config
+	allErrs = append(allErrs, configvalidation.ValidateGardenletConfiguration(gardenletConfig, fldPath)...)
+
+	if gardenletConfig.SeedConfig != nil {
```

Comment:
I find it a bit confusing, that the `seedConfig`/`seedTemplate` of a `ManagedSeed` is validated, but the the `seedConfig` of the standalone `GardenletConfiguration` is not validated. Did I miss something here?

Should we move the basic validation of the `SeedTemplate` to `corevalidation` and then call it from the `seedmanagement` validation and `gardenletconfig` validation? Additional checks can then be performed in both packages like enforcing an empty `Name` or non-empty `Name`.

Reply:
Yes, a `seedConfig` in a standalone `GardenletConfiguration` (outside of a `ManagedSeed`) is not validated. This is the status quo, I am not sure if we really want to change it at this point. If we go this way, then perhaps we should also use the `SeedTemplate` in both (see also https://github.com/gardener/gardener/pull/3414#discussion_r563792340) - if we are doing disruptive changes,  then let's at least change the design to a cleaner one. But as said above, I am not a big fan of such changes of this point.

Reply:
> If we go this way, then perhaps we should also use the `SeedTemplate` in both (see also [#3414 (comment)](https://github.com/gardener/gardener/pull/3414#discussion_r563792340)) - if we are doing disruptive changes, then let's at least change the design to a cleaner one.

Agree, that would be also my preferred approach.

> But as said above, I am not a big fan of such changes of this point.

Also agree. I just wanted to double check, that I didn't miss anything here. 
Let's set this straight in a dedicated PR (is also a breaking change, so it definitely should be in its own PR). Can you open an issue for this or just open a PR afterwards?

Reply:
OK, I will do it, I already have it on my list from the previous review.